aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -0,0 +1,2326 @@
+        add_stake_events: EventHandle<AddStakeEvent>,
+        reactivate_stake_events: EventHandle<ReactivateStakeEvent>,
+        unlock_stake_events: EventHandle<UnlockStakeEvent>,
+        withdraw_stake_events: EventHandle<WithdrawStakeEvent>,
+    }
+
+    struct AddStakeEvent has drop, store {
+        pool_address: address,
+        delegator_address: address,
+        amount_added: u64,
+        add_stake_fee: u64,
+    }
+
+    struct ReactivateStakeEvent has drop, store {
+        pool_address: address,
+        delegator_address: address,
+        amount_reactivated: u64,
+    }
+
+    struct UnlockStakeEvent has drop, store {
+        pool_address: address,
+        delegator_address: address,
+        amount_unlocked: u64,
+    }
+
+    struct WithdrawStakeEvent has drop, store {
+        pool_address: address,
+        delegator_address: address,
+        amount_withdrawn: u64,
+    }
+
+    #[view]
+    /// Return whether supplied address `addr` is owner of a delegation pool.
+    public fun owner_cap_exists(addr: address): bool {
+        exists<DelegationPoolOwnership>(addr)
+    }
+
+    #[view]
+    /// Return address of the delegation pool owned by `owner` or fail if there is none.
+    public fun get_owned_pool_address(owner: address): address acquires DelegationPoolOwnership {
+        assert_owner_cap_exists(owner);
+        borrow_global<DelegationPoolOwnership>(owner).pool_address
+    }
+
+    #[view]
+    /// Return whether a delegation pool exists at supplied address `addr`.
+    public fun delegation_pool_exists(addr: address): bool {
+        exists<DelegationPool>(addr)
+    }
+
+    #[view]
+    /// Return the index of current observed lockup cycle on delegation pool `pool_address`.
+    public fun observed_lockup_cycle(pool_address: address): u64 acquires DelegationPool {
+        assert_delegation_pool_exists(pool_address);
+        borrow_global<DelegationPool>(pool_address).observed_lockup_cycle.index
+    }
+
+    #[view]
+    /// Return the unique observed lockup cycle where delegator `delegator_address` may have
+    /// unlocking (or already unlocked) stake to be withdrawn from delegation pool at `pool_address`.
+    public fun has_pending_withdrawal(pool_address: address, delegator_address: address): (bool, u64) acquires DelegationPool {
+        assert_delegation_pool_exists(pool_address);
+        let pool = borrow_global<DelegationPool>(pool_address);
+        let (withdrawal_exists, withdrawal_olc) = pending_withdrawal_exists(pool, delegator_address);
+        (withdrawal_exists, withdrawal_olc.index)
+    }
+
+    #[view]
+    /// Return the stake amount on the pending withdrawal of regular delegator `delegator_address`
+    /// on `pool_address` and whether it can be executed (is inactive) or not.
+    public fun get_pending_withdrawal(pool_address: address, delegator_address: address): (bool, u64) acquires DelegationPool {
```

Comment:
Why not just return the amount in this function? has_pending_withdrawal already returns whether the user has any pending withdrawal at all.

Reply:
you mean `has_pending_withdrawal` could be removed as delegators would only use `get_pending_withdrawal` to detect if they have any unlocking/unlocked stake already? Knowing the OLC of the pending withdrawal is not something a delegator would be interested in, but rather if they can withdraw and how much stake.

Reply:
Makes sense thanks for the clarification